tests/tools/fio/fio.go

```diff
@@ -0,0 +1,126 @@
+// Package fio wraps calls to the fio tool.
+// It assumes the tool is executable by "fio", but
+// gives the option to specify another executable
+// path by setting environment variable FIO_EXE.
+package fio
+
+import (
+	"io/ioutil"
+	"log"
+	"os"
+	"os/exec"
+	"strings"
+	"sync"
+)
+
+// List of fio flags
+const (
+	JobNameFlag = "--name"
+)
+
+// Runner is a helper for running fio commands
+type Runner struct {
+	Exe     string
+	DataDir string
+	Global  Config
+}
+
+// NewRunner creates a new fio runner
+func NewRunner() (*Runner, error) {
+	Exe := os.Getenv("FIO_EXE")
+	if Exe == "" {
+		Exe = "fio"
+	}
+
+	dataDir, err := ioutil.TempDir("", "fio-data")
+	if err != nil {
+		return nil, err
+	}
+
+	return &Runner{
+		Exe:     Exe,
+		DataDir: dataDir,
+		Global: Config{
+			{
+				Name: "global",
+				Options: map[string]string{
+					"openfiles":         "10",
+					"create_fsync":      "0",
+					"create_serialize":  "1",
+					"file_service_type": "sequential",
+					"ioengine":          "libaio",
+					"direct":            "1",
+					"iodepth":           "32",
+					"blocksize":         "1m",
+					"refill_buffers":    "",
+					"rw":                "write",
+					"unique_filename":   "1",
+					"directory":         dataDir,
+				},
+			},
+		},
+	}, nil
+}
+
+// Cleanup cleans up the data directory
+func (fr *Runner) Cleanup() {
+	if fr.DataDir != "" {
+		os.RemoveAll(fr.DataDir) //nolint:errcheck
+	}
+}
+
+// RunConfigs runs fio using the provided Configs
+func (fr *Runner) RunConfigs(cfgs ...Config) (stdout, stderr string, err error) {
+	var args []string
+
+	// Apply global config before any other configs
+	for _, cfg := range append(append([]Config{}, fr.Global), cfgs...) {
```

Comment:
Is there any benefit for doing the double `append` here? Can it be simplified as

```suggestion
	for _, cfg := range append([]Config{fr.Global}, cfgs...) {
```

Reply:
you're right; I think I was copying a pattern from `*CliTest.Run` but it's actually not needed here.